integration/src/test/java/org/mule/test/components/tracing/AbstractTracingIntegrationTestCase.java

```diff
@@ -0,0 +1,196 @@
+  private final int port;
+  private final String path;
+  private final boolean secure;
+
+  protected GenericContainer<?> collector;
+
+  @ClassRule
+  public static SelfSignedCertificateRule serverTls = new SelfSignedCertificateRule();
+
+  @ClassRule
+  public static SelfSignedCertificateRule clientTls = new SelfSignedCertificateRule();
+
+  @ClassRule
+  public static final TestGrpcServerRule server = new TestGrpcServerRule();
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return asList(new Object[][] {
+        {"GRPC", "http://", COLLECTOR_OTLP_GRPC_PORT, "", false},
+        {"HTTP", "http://", COLLECTOR_OTLP_HTTP_PORT, "/v1/traces", false},
+        {"GRPC", "https://", COLLECTOR_OTLP_GRPC_MTLS_PORT, "", true},
+        {"HTTP", "https://", COLLECTOR_OTLP_HTTP_MTLS_PORT, "/v1/traces", true}
+    });
+  }
+
+  public AbstractTracingIntegrationTestCase(String exporterType, String schema, int port, String path, boolean secure) {
+    this.exporterType = exporterType;
+    this.schema = schema;
+    this.port = port;
+    this.path = path;
+    this.secure = secure;
+  }
+
+  @Before
+  public void before() {
+    exposeHostPorts(server.httpPort());
+    // Configuring the collector test-container
+    collector =
+        new GenericContainer<>(COLLECTOR_IMAGE)
+            .withImagePullPolicy(PullPolicy.alwaysPull())
+            .withEnv("LOGGING_EXPORTER_LOG_LEVEL", "INFO")
+            .withCopyFileToContainer(
+                                     forHostPath(serverTls.certificateFile().toPath(), 365),
+                                     "/server.cert")
+            .withCopyFileToContainer(
+                                     forHostPath(serverTls.privateKeyFile().toPath(), 365), "/server.key")
+            .withCopyFileToContainer(
+                                     forHostPath(clientTls.certificateFile().toPath(), 365),
+                                     "/client.cert")
+            .withEnv("MTLS_CLIENT_CERTIFICATE", "/client.cert")
+            .withEnv("MTLS_SERVER_CERTIFICATE", "/server.cert")
+            .withEnv("MTLS_SERVER_KEY", "/server.key")
+            .withEnv(
+                     "OTLP_EXPORTER_ENDPOINT", "host.testcontainers.internal:" + server.httpPort())
+            .withClasspathResourceMapping(
+                                          "otel.yaml", "/otel.yaml", READ_ONLY)
+            .withCommand("--config", "/otel.yaml")
+            .withExposedPorts(
+                              COLLECTOR_OTLP_GRPC_PORT,
+                              COLLECTOR_OTLP_HTTP_PORT,
+                              COLLECTOR_OTLP_GRPC_MTLS_PORT,
+                              COLLECTOR_OTLP_HTTP_MTLS_PORT,
+                              COLLECTOR_HEALTH_CHECK_PORT)
+            .waitingFor(Wait.forHttp("/").forPort(COLLECTOR_HEALTH_CHECK_PORT));
+
+    collector.start();
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENABLED, TRUE.toString());
```

Comment:
Can we use the SystemProperty rule?

Reply:
I want to set some system properties which dynamically change according to the collector and some of them have to be set depending on the different parameters. I don't think I can control that maintaining the test as parameterized.